index.d.ts

```diff
@@ -33,6 +33,7 @@ export type DecamelizeKeys<
 	Separator extends string = '_',
 	Exclude extends readonly unknown[] = EmptyTuple,
 	Deep extends boolean = false,
+	StopPaths extends readonly string[] = EmptyTuple,
```

Comment:
You are not actually using the `StopPaths` option in the types.

https://github.com/sindresorhus/camelcase-keys/blob/27e1e9dd25a865a911aaa9eb16cfa018438fc4b3/index.d.ts#L68

You also need to add a type test in https://github.com/sindresorhus/decamelize-keys/blob/main/index.test-d.ts